Tests/MapboxMapsTests/Foundation/MapViewTests.swift

```diff
@@ -252,23 +252,60 @@ final class MapViewTests: XCTestCase {
 
         XCTAssertEqual(displayLink.$isPaused.setStub.invocations.map(\.parameters), [true])
     }
 
+    func testDisplayLinkPausedWhenAppDidEnterBackground() {
+        displayLink.$isPaused.setStub.reset()
+
+        notificationCenter.post(name: UIApplication.didEnterBackgroundNotification, object: nil)
+
+        XCTAssertEqual(displayLink.$isPaused.setStub.invocations.map(\.parameters), [true])
+    }
+
     func testReleaseDrawablesInvokedWhenAppDidBecomeActive() {
         notificationCenter.post(name: UIApplication.didEnterBackgroundNotification, object: nil)
 
         XCTAssertEqual(metalView.releaseDrawablesStub.invocations.count, 1)
     }
 
-    func testDisplayLinkResumedWhenAppDidBecomeActive() {
+    func testDisplayLinkResumedWhenAppDidBecomeActiveOnIOS12() throws {
+        if #available(iOS 13.0, *) {
+            throw XCTSkip("Test applies only on iOS 12")
+        }
+
         notificationCenter.post(name: UIApplication.didBecomeActiveNotification, object: nil)
 
         XCTAssertEqual(displayLink.$isPaused.setStub.invocations.map(\.parameters), [false])
     }
 
+    func testSubscribesToCorrectNotifications() {
+        let observers = notificationCenter.addObserverStub.invocations.map(\.parameters.observer)
+
+        XCTAssertTrue(observers.allSatisfy { ($0 as AnyObject) === mapView })
+
+        if #available(iOS 13.0, *) {
+            XCTAssertEqual(notificationCenter.addObserverStub.invocations.count, 6)
+            XCTAssertEqual(notificationCenter.addObserverStub.invocations[0].parameters.name, UIScene.didEnterBackgroundNotification)
+            XCTAssertEqual(notificationCenter.addObserverStub.invocations[1].parameters.name, UIScene.willDeactivateNotification)
+            XCTAssertEqual(notificationCenter.addObserverStub.invocations[2].parameters.name, UIScene.didActivateNotification)
+            XCTAssertEqual(notificationCenter.addObserverStub.invocations[3].parameters.name, UIApplication.didEnterBackgroundNotification)
+            XCTAssertEqual(notificationCenter.addObserverStub.invocations[4].parameters.name, UIApplication.willResignActiveNotification)
+            XCTAssertEqual(notificationCenter.addObserverStub.invocations[5].parameters.name, UIApplication.didReceiveMemoryWarningNotification)
+        } else {
+            XCTAssertEqual(notificationCenter.addObserverStub.invocations.count, 4)
+
+            XCTAssertEqual(notificationCenter.addObserverStub.invocations[0].parameters.name, UIApplication.didBecomeActiveNotification)
+            XCTAssertEqual(notificationCenter.addObserverStub.invocations[1].parameters.name, UIApplication.didEnterBackgroundNotification)
+            XCTAssertEqual(notificationCenter.addObserverStub.invocations[2].parameters.name, UIApplication.willResignActiveNotification)
+            XCTAssertEqual(notificationCenter.addObserverStub.invocations[3].parameters.name, UIApplication.didReceiveMemoryWarningNotification)
+        }
+    }
```

Comment:
[discussion] May be it makes sense to split this method into two – one for iOS13+ and another one for iOS12-? 
My points are that a test method should be simple and third-party tools may use the test names to detect flaky tests, for example. And the `if` statement for XCTAsserts doesn't work well for such a configuration.
WDYT?

Reply:
Agree, separated these two cases into separate tests.